packages/share_plus/share_plus/lib/src/share_plus_linux.dart

```diff
@@ -40,10 +39,12 @@ class SharePlusLinuxPlugin extends SharePlatform {
           .join('&'),
     );
 
-    if (await urlLauncher.canLaunch(uri.toString())) {
-      await urlLauncher.launchUrl(uri.toString(), const LaunchOptions());
-    } else {
-      throw Exception('Unable to share on web');
+    final launchResult = await urlLauncher.launchUrl(
+      uri.toString(),
+      const LaunchOptions(),
+    );
+    if (!launchResult) {
+      throw Exception('Failed to launch mailto: URI');
```

Comment:
As commented, the launchUrl can either return false or fail with an exception, in the case of returning false we throw an Exception.

I did not use `PlatformException` here because I don't think we should be creating them outside platform code, so I think just a generic `Exception` is fine here.

Reply:
Yeah, it makes sense.